storymap_helper.py

```diff
@@ -54,53 +64,313 @@ def __init__(
         with open(_storymap_template_filepath, "r") as _storymap_template_file:
             self._storymap_template = yaml.safe_load(_storymap_template_file)
 
+        # Replace placeholders in the template
+        self._replace_placeholders()
+
+        # Create the StoryMap object with the title
         self._storymap = self.create_new_storymap()
 
-    def _populate_storymap_cover(self):
-        my_cover_image = Image(
-            path="https://story.maps.arcgis.com/sharing/rest/content/items/2fcc801c983a402eb427dd5cd07ee759/data"
-        )
-        self._storymap.cover(
-            type="full",
-            title="Storymap for {design_name}".format(
+    def _replace_placeholders(self):
+        """Replace placeholders in the YAML template with actual values."""
+ 
+        placeholders = {
```

Comment:
It would be great if we had a dataclass for this for e.g. so we are not creating arbitrary placeholder, perhaps somewhere that defines all the placeholders in the YAML

Reply:
Refactored part to YAML section